src/slicegpt/hf_utils.py

```diff
@@ -86,12 +86,15 @@ def get_model_and_tokenizer(
     else:
         model_type = "pretrained"
 
-    # sliced model is always a local model
     if model_path is None:
         local_model = False
     else:
         local_model = True
 
+    if uninitialized:
+        # Sliced models always have a model_path. Check if config.json exists, then it's a local sliced model
+        local_model = pathlib.Path(f"{model_path}/config.json").exists()
```

Comment:
Currently when we save a sliced model with `--save-dir sliced_models` we get the following two files: 
- `sliced_models/opt-1.3b_0.25.json`
- `sliced_models/opt-1.3b_0.25.pt`

Should we change the logic to save files like `sliced_models/model.pt` and `sliced_models/config.json` ?

Pros: probably more HF-like
Cons: probably could do with a minor change to the saving logic to create the two files in a folder within sliced_models which makes it clear what the model is, e.g. `sliced_models/opt-1.3b-0.25-slicing/`.

Let me know what you think @PashminaCameron

Reply:
The json we save is not the HF config json, right? So that would be slightly confusing.

We could save the HF config as well, and include our rotation config in it or save it as a separate file, but not sure if either approach is helpful since we need to go to HF to instantiate a model anyway.

Reply:
The json for sliced models is not called config.json, it's called model_name_without_ext.json. Saving config.json etc - we can get to it when we tackle the loading/saving in HF format properly.

Reply:
To be clear, I am proposing not mixing up the slicing related files with HF base model files at all.

Reply:
I understand and completely agree with proposal, makes a lot of sense! The issue is that the above code fails - we can't load a sliced model with this because a `config.json` is never created by saving a sliced model.

Reply:
For local models, it's the user's responsibility to copy the config.json to the sliced_model_path, so it will exist or loading will throw NotImplementedError and then the user needs to fix the issue. For HF Hub supported models, we will load the sliced model based on the config.json from HF hub. 

So the only issue is that we are not making it easy for a local model user by copying config.json over to the --save-dir specified when slicing models, but that would also apply to other files such as the tokenizer files (and there can be an unspecified number of those), so where do we stop? We _could_ copy all *.json files when slicing local models.

Reply:
Above makes sense, thanks for clarifying! I've found the source of my confusion: loading a sliced HF model will lead to this line getting executed: `logging.info(f"Loading {model_type} {model_name} from Hugging Face (cache)")`. We should make sure this doesn't happen as this line should be used to inform that weights are getting pulled from HF, which we avoid when loading a sliced checkpoint as it is expensive and superfluous as any loaded weights would get replaced by those from the chkpt.

 I would suggest removing https://github.com/microsoft/TransformerCompression/blob/2ec761b29b6ab654c098484e4114359faaa7315f/experiments/run_slicegpt_perplexity.py#L147 and updating the logging in the current file.

Reply:
Copying all config and tokenizer files for local sliced models now.

Reply:
Updated logging to clarify config/weights being loaded from Hub vs local path.